instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/InstrumenterBuilder.java

```diff
@@ -199,6 +199,17 @@
         InstrumenterConstructor.propagatingToDownstream(setter), SpanKindExtractor.alwaysClient());
   }
 
+  /**
+   * Returns a new {@link Instrumenter} which will create client spans and inject context into
+   * requests.
+   */
+  public Instrumenter<REQUEST, RESPONSE> newClientInstrumenter(
+      TracingBuilder<REQUEST, RESPONSE, ?> tracingBuilder, TextMapSetter<REQUEST> setter) {
```

Comment:
The idea is to remove the methods that don't accept a `TracingBuilder`

Reply:
Should we really do that? Requiring a `TracingBuilder` will make it harder to use `Instrumenter` ad-hoc - for example in our own javaagent instrumentations (or in libs like spring sleuth, probably).

Reply:
Good point. I struggled to come up with a pattern for applying the settings to the Instrumenter that isn't easy to forget to do, e.g. a protected `apply` method.

Any idea come to mind?

Reply:
I mean, adding new `newInstrumenter()` methods that accept a `TracingBuilder` seems fine, it's the removing idea that might be problematic; I don't really have better ideas that this. `copySettingsFrom(TracingBuilder)`?